### PR TITLE
FOQUS UI Local/Cloud Updates/Bug Fixes

### PR DESCRIPTION
**Pull request by boverhof**
## Fixes/Addresses:
- Works with TurbineClient TAG/release 4.0.1
- BFB flowsheet/ACM works with new FOQUS Cloud and LOCAL deployment 
- FOQUS Cloud tested Optimization NGCC_EEMPA

## Issues Fixed
- https://github.com/CCSI-Toolset/FOQUS/issues/1010
- https://github.com/CCSI-Toolset/turb_client/issues/19
- 
## Summary/Motivation:
- Involves large scale changes across several projects turb_client, FOQUS, and FOQUS Cloud.
- Necessary for operation of FOQUS Cloud and SSL Versioning with AWS API Gateway

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the copyright and license terms described in the LICENSE.md file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by lbianchi-lbl** [comment]
@boverhof will look at this in more detail, but a general question: does this need updated version of other packages/repositories? if so, the appropriate versions/git refa should be specified in `setup.py`.

**Comment by boverhof** [comment]
@lbianchi-lbl only turb_client 4.0.1, which isn't "released"

**Comment by boverhof** [comment]
FOQUS Cloud tested Optimization NGCC_EEMPA

Screen Shot 2022-08-25 at 3 09 24 PM

**Comment by lbianchi-lbl** [comment]
> @lbianchi-lbl only turb_client 4.0.1, which isn't "released"

@boverhof For testing purpose (i.e. it will have to be changed before we need to create a PyPI package for FOQUS next month), you can specify a GitHub repo URL pointing to any valid existing ref (branch, tag, partial commit hash with 7+ chars) as part of the requirement in `setup.py`: e.g., if you need to install the `boverhof_08_25_2022` branch:

```py
setup(
     ...
     install_requires=[
          ...,
          "TurbineClient @ https://github.com/CCSI-Toolset/turb_client/archive/boverhof_08_25_2022.zip",
          ...
     ]
)
```

**Comment by boverhof** [comment]
@lbianchi-lbl I have updated the FOQUS pull request with the change you specified above

File: foqus_lib/framework/graph/graph.py [hunk 1]
```diff
@@ -741,6 +735,7 @@ def solveListValTurbine(self, valueList=None, maxSend=20, sid=None, jobIds=[]):
                 )
                 rp += 1
                 for job in jres:
+                    assert type(job) is dict
```
Comment (lbianchi-lbl) [inline foqus_lib/framework/graph/graph.py:738]:
```suggestion
                    assert isinstance(job, dict)
```

File: foqus_lib/framework/graph/graph.py [hunk 2]
```diff
@@ -785,6 +780,7 @@ def solveListValTurbine(self, valueList=None, maxSend=20, sid=None, jobIds=[]):
                                     "graphError": self.res_fin[i],
                                 }
                             else:
+                                assert type(self.res[i]) is dict
```
Comment (lbianchi-lbl) [inline foqus_lib/framework/graph/graph.py:783]:
```suggestion
                                assert isinstance(self.res[i], dict)
```

File: foqus_lib/gui/main/mainWindow.py [hunk 3]
```diff
@@ -1223,36 +1258,50 @@ def runSim(self, node=None, valList=None, rows=None):
...
         node given by the name is evaluated
         """
+        logging.getLogger("foqus." + __name__).debug("Start Simulation")
         turb_config = self.dat.flowsheet.turbConfig
         turb_sim_list = self.dat.flowsheet.turbineSimList()
         self.applyNodeEdgeChanges()
         if node in self.dat.flowsheet.nodes:
             self.dat.flowsheet.onlySingleNode = node
             self.setStatus("Running Single Node ({0}) Simulation...".format(node))
+            logging.getLogger("foqus." + __name__).debug(
```
Comment (lbianchi-lbl) [inline foqus_lib/gui/main/mainWindow.py:1268]:
```suggestion
            _log.debug(
```

File: foqus_lib/gui/main/mainWindow.py [hunk 4]
```diff
@@ -1223,36 +1258,50 @@ def runSim(self, node=None, valList=None, rows=None):
...
+                "Running Single Node ({0}) Simulation...".format(node)
+            )
         elif valList is not None:
             self.dat.flowsheet.onlySingleNode = None
             self.setStatus("Running Flowsheet Set...")
+            logging.getLogger("foqus." + __name__).debug(
```
Comment (lbianchi-lbl) [inline foqus_lib/gui/main/mainWindow.py:1274]:
```suggestion
            _log.debug(
```

File: foqus_lib/gui/main/mainWindow.py [hunk 5]
```diff
@@ -1223,36 +1258,50 @@ def runSim(self, node=None, valList=None, rows=None):
...
+                "Running Flowsheet Set: %s", valList
+            )
         else:
             self.dat.flowsheet.onlySingleNode = None
             self.setStatus("Running Single Flowsheet Simulation...")
+            logging.getLogger("foqus." + __name__).debug(
```
Comment (lbianchi-lbl) [inline foqus_lib/gui/main/mainWindow.py:1280]:
```suggestion
            _log.debug(
```

File: foqus_lib/gui/main/mainWindow.py [hunk 6]
```diff
@@ -1223,36 +1258,50 @@ def runSim(self, node=None, valList=None, rows=None):
...
+                "Running Single Flowsheet Simulation..."
+            )
         if self.dat.foqusSettings.runFlowsheetMethod == 0:
             # run in FOQUS
             if valList is not None:
+                logging.getLogger("foqus." + __name__).debug("Local multiRun")
```
Comment (lbianchi-lbl) [inline foqus_lib/gui/main/mainWindow.py:1286]:
```suggestion
                _log.debug("Local multiRun")
```

File: foqus_lib/gui/main/mainWindow.py [hunk 7]
```diff
@@ -1223,36 +1258,50 @@ def runSim(self, node=None, valList=None, rows=None):
...
                 self.multiRun = self.dat.flowsheet.runListAsThread(valList)
                 self.singleRun = None
             else:
+                logging.getLogger("foqus." + __name__).debug("Local singleRun")
```
Comment (lbianchi-lbl) [inline foqus_lib/gui/main/mainWindow.py:1290]:
```suggestion
                _log.debug("Local singleRun")
```

File: foqus_lib/gui/main/mainWindow.py [hunk 8]
```diff
@@ -1223,36 +1258,50 @@ def runSim(self, node=None, valList=None, rows=None):
...
                 self.singleRun = self.dat.flowsheet.runAsThread()
                 self.multiRun = None
         elif self.dat.foqusSettings.runFlowsheetMethod == 1:
             # Submit to Turbine <-> FOQUS consumer
             # first save a session file (need to upload to turbine)
             self.dat.flowsheet.uploadFlowseetToTurbine(dat=self.dat, reset=False)
             if valList is not None:
+                logging.getLogger("foqus." + __name__).debug("FOQUS Cloud multiRun")
```
Comment (lbianchi-lbl) [inline foqus_lib/gui/main/mainWindow.py:1298]:
```suggestion
                _log.debug("FOQUS Cloud multiRun")
```

File: foqus_lib/gui/main/mainWindow.py [hunk 9]
```diff
@@ -1223,36 +1258,50 @@ def runSim(self, node=None, valList=None, rows=None):
...
                 self.multiRun = self.dat.flowsheet.runListAsThread(
                     valList, useTurbine=True
                 )
                 self.singleRun = None
             else:
+                logging.getLogger("foqus." + __name__).debug("FOQUS Cloud singleRun")
```
Comment (lbianchi-lbl) [inline foqus_lib/gui/main/mainWindow.py:1304]:
```suggestion
                _log.debug("FOQUS Cloud singleRun")
```

File: foqus_lib/gui/main/mainWindow.py [hunk 10]
```diff
@@ -1364,6 +1415,9 @@ def stopSim(self):
             self.nodeDock.runButton.setEnabled(True)
             self.nodeDock.stopButton.setEnabled(False)
             if self.singleRun.res[0]:
+                logging.getLogger("foqus." + __name__).debug(
```
Comment (lbianchi-lbl) [inline foqus_lib/gui/main/mainWindow.py:1418]:
```suggestion
                _log.debug(
```

**Review by lbianchi-lbl — COMMENTED** [review]
From a quick look this looks OK. I've added a few very minor comments on the Python code with suggestions. @boverhof if you decide to accept them, you should be able to commit them in one or more batches directly from the GitHub UI without needing to edit the code locally.

**Review by lbianchi-lbl — APPROVED** [review]

**Pull request merged**